runtime/lua/vim/treesitter.lua

```diff
@@ -137,4 +137,45 @@ function M.is_ancestor(dest, source)
   return false
 end
 
+--- Get the next node of the same parent
+---
+---@param node The start node
+---@param opts table A table of optional arguments:
+---                  - switch_parent: (boolean, default false) Allow to return the first node from
+---                  the next parent if the start node is the last node of its parent.
+---                  - return_next_parent: (boolean, default false) With switch_parent = true, allow
+---                  to return the next parent as the next node if it has no children.
+---
+---@returns The next node
+function M.get_next_node(node, opts)
```

Comment:
I used an opts table here as it seems like something I see a lot in other APIs when a function has optional parameters, good or bad idea ?

Reply:
Good, in general. It's most used for API functions with stability guarantees, but widely preferred for, well, optional arguments.

(Rule of thumb: mandatory is positional, optional keyword style -- which in Lua means part of an opts table)